### PR TITLE
feat: Add CouchDB implementation for pvtdata storage

### LINKED ISSUE #38: Add CouchDB implementation for pvtdata storage

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

closes #38

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ run:
   issues-exit-code: 1
   tests: true
   build-tags: ["testing"]
-  skip-dirs:  [""]
+  skip-dirs:  ["pvtdatastorage/common"]
```
Comment (troyronda) [inline .golangci.yml]: Why this skip?

File: pkg/pvtdatastorage/cdbpvtdatastore/couchdb_conv.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
+package pvtdatastorage
```
Comment (troyronda) [inline pkg/pvtdatastorage/cdbpvtdatastore/couchdb_conv.go]: Copyright header is missing.

File: pkg/pvtdatastorage/cdbpvtdatastore/test_exports.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
```
Comment (troyronda) [inline pkg/pvtdatastorage/cdbpvtdatastore/test_exports.go]: Is it correct that SecureKey isn't also listed? Did we make any non-trivial modifications to this file?

File: pkg/pvtdatastorage/common/collelgproc.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
```
Comment (troyronda) [inline pkg/pvtdatastorage/common/collelgproc.go]: Is it correct that SecureKey isn't also listed? Did we make any non-trivial modifications to this file?

File: pkg/pvtdatastorage/common/helper.go [hunk 5]
```diff
@@ -0,0 +1,235 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
```
Comment (troyronda) [inline pkg/pvtdatastorage/common/helper.go]: Is it correct that SecureKey isn't also listed? Did we make any non-trivial modifications to this file?

File: pkg/pvtdatastorage/common/kv_encoding.go [hunk 6]
```diff
@@ -0,0 +1,192 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
```
Comment (troyronda) [inline pkg/pvtdatastorage/common/kv_encoding.go]: Is it correct that SecureKey isn't also listed? Did we make any non-trivial modifications to this file?

File: pkg/pvtdatastorage/common/v11.go [hunk 7]
```diff
@@ -0,0 +1,78 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
```
Comment (troyronda) [inline pkg/pvtdatastorage/common/v11.go]: Is it correct that SecureKey isn't also listed? Did we make any non-trivial modifications to this file?

File: .golangci.yml (hunk 1 above)
Comment (fqutishat) [reply .golangci.yml]: this code is copied from fabric

File: pkg/pvtdatastorage/cdbpvtdatastore/couchdb_conv.go (hunk 2 above)
Comment (fqutishat) [reply pkg/pvtdatastorage/cdbpvtdatastore/couchdb_conv.go]: fixed in the next pr

File: pkg/pvtdatastorage/cdbpvtdatastore/test_exports.go (hunk 3 above)
Comment (fqutishat) [reply pkg/pvtdatastorage/cdbpvtdatastore/test_exports.go]: fixed in the next pr

File: pkg/pvtdatastorage/common/collelgproc.go (hunk 4 above)
Comment (fqutishat) [reply pkg/pvtdatastorage/common/collelgproc.go]: fixed in the next pr

File: pkg/pvtdatastorage/common/helper.go (hunk 5 above)
Comment (fqutishat) [reply pkg/pvtdatastorage/common/helper.go]: fixed in the next pr

File: pkg/pvtdatastorage/common/kv_encoding.go (hunk 6 above)
Comment (fqutishat) [reply pkg/pvtdatastorage/common/kv_encoding.go]: fixed in the next pr

File: pkg/pvtdatastorage/common/v11.go (hunk 7 above)
Comment (fqutishat) [reply pkg/pvtdatastorage/common/v11.go]: fixed in the next pr

File: .golangci.yml (hunk 1 above)
Comment (troyronda) [reply .golangci.yml]: doesn't mean we should also skip :)

**Pull request merged**